src/js/components/uploadDetachedFiles/UploadDetachedFileValidation.jsx

```diff
@@ -188,6 +190,43 @@ export default class UploadDetachedFileValidation extends React.Component {
 	// 2: Fetching file metadata failed
 	// 3: File already has been submitted in another submission
 
+	uploadFileHelper(local, submission){
+		if(local){
+			return UploadHelper.performDetachedLocalCorrectedUpload(submission);
+		}
+		return UploadHelper.performDetachedFileCorrectedUpload(submission);
+	}
+
+	uploadFile(item) {
+		// upload specified file
+		this.props.setSubmissionState('uploading');
+		let submission = this.props.submission;
+		submission.files.detached_award = this.state.detachedAward;
+		submission.files.detached_award.file = item;
+		submission.sub = this.state.submissionID;
+		submission.meta['startDate'] = this.state.rep_start;
+		submission.meta['endDate'] = this.state.rep_end;
+		submission.meta['subTierAgency'] = this.state.agency;
+
+		this.uploadFileHelper(kGlobalConstants.LOCAL === true && !this.isUnmounted, submission)
```

Comment:
My main point was that `!this.isUnmounted` has absolutely nothing to do with testing if it's a local upload or not, it shouldn't be being used for it.